src/headless/plugins/chat/model.js

```diff
@@ -731,6 +731,7 @@ const ChatBox = ModelWithContact.extend({
                 'id': origin_id,
                 'xmlns': Strophe.NS.FASTEN
             }).c('retract', {xmlns: Strophe.NS.RETRACT})
+            .c('fallback', {xmlns: Strophe.NS.FALLBACK}).c('body').t("This person attempted to retract a previous message, but it's unsupported by your client.").up();
         return _converse.connection.send(msg);
```

Comment:
Tests are still failing because you're still creating the `body` inside the `fallback`.

It needs to be one level up.
https://github.com/conversejs/converse.js/runs/5598110420?check_suite_focus=true#step:4:3087

You first need to call `up()` before calling `c.('body')`, like explained in my previous comment.

Reply:
Just realized the purpose of  `Prototype.up()` function, you are right.

Reply:
I made another mistake, just a sec

Reply:
This is a nitpick, but can you please put a line-break before the `c('body')`?

For example:

```javascript
const msg = $msg({
                'id': u.getUniqueId(),
                'to': this.get('jid'),
                'type': "chat"
            })
            .c('store', {xmlns: Strophe.NS.HINTS}).up()
            .c("apply-to", {
                'id': origin_id,
                'xmlns': Strophe.NS.FASTEN
            }).c('retract', {xmlns: Strophe.NS.RETRACT}).up()
            .c('fallback', {xmlns: Strophe.NS.FALLBACK}).up()
            .c('body').t("This person attempted to retract a previous message, but it's unsupported by your client.");
```

Reply:
already did, the latest commit adds that